src/pages/postgraphile/jwk-verification.md

```diff
@@ -201,6 +201,29 @@ appropriate and succinct response:
 }
 ```
 
+If, however, you prefer to assign a default PostgreSQL role when no token has been provided (in a way replicating `pgDefaultRole` behavior) then that's also possible by intercepting when `err.code === 'credentials_required'`:
+
+```javascript
+const authErrors = (err, req, res, next) => {
+  if (err.name === "UnauthorizedError") {
+    console.log(err); // You will still want to log the error...
+    if (err.code === 'credentials_required') {
+      // No token provided, so assign a default PostgreSQL role (see pgDefaultRole) and continue
+      req.user = {
+        role: 'unauthenticated'
+      }
```

Comment:
I would set `req.user = null` here (there is no user) and interpret the null result in pgSettings

Reply:
Perhaps the express-jwt option `credentialsRequired: false` would achieve this in less code?

```js
app.use(jwt({
  secret: 'hello world !',
  credentialsRequired: false
}));
```

Reply:
Doh! Missed `credentialsRequired`.... yeah, that's much better!